### PR TITLE
MOODLE-6106

### PR DESCRIPTION
**Pull request by sampraxis**
Fixed plugin unintended copy badges from a course, when user copy an activity and a section.

### CODE REVIEW

**Review by frederikmillingpytlick — CHANGES_REQUESTED** [review]
One suggestion and perhaps you could take a look at the GitHub action - one of them are failing

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            moodle: MOODLE_311_STABLE
+            plugin_branch: MOODLE_311_STABLE
+
+    steps:
+      - name: Setting up DB mysql
+        if: ${{ matrix.db == 'mysqli' }}
+        uses: johanmeiring/mysql-action@tmpfs-patch
+        with:
+          collation server: utf8mb4_unicode_ci
+          mysql version: 5.7
```
[the remark this answers is not in the document]
Comment (sampraxis) [reply .github/workflows/push.yml:39]: I tried it with moodle 3.10 to run with mysql 8 but it failed.
It is most likely PHP 7.3 with moodle 3.10 ain't compatible with mysql 8 driver.

[the remark this answers is not in the document]
Comment (sampraxis) [reply .github/workflows/push.yml:39]: Here is the log of error
https://github.com/praxisdigital/moodle-block_sharing_cart/runs/5331006778?check_suite_focus=true#step:8:172

**Review by frederikmillingpytlick — APPROVED** [review]
:rocket:

**Review by frederikmillingpytlick — APPROVED** [review]
:rocket: :rocket: :rocket: :rocket: :rocket: :rocket:

**Pull request merged**